[Feature] User group permission

roycephan opened the pull request:
#### Status

READY

#### What's this PR do?

- [x] auth user's permission & group permission
- [x] test for group permission

hieunmce commented on backend/auth/auth.go:
```diff
@@ -62,14 +63,9 @@ func Authorization(cfg *backendConfig.Config) func(next http.Handler) http.Handl
-				return
-			}
-			// POST: localhost:2999/databases/fortress/users/create
+			// POST: localhost:2999/databases/fortress/table/users/create
```
we should comment more correct or comment with explanation

hieunmce commented on backend/auth/auth.go:
```diff
@@ -92,51 +88,50 @@ func Authorization(cfg *backendConfig.Config) func(next http.Handler) http.Handl
-					//fortress
-					dbName = uriParts[2]
-					//users
+					dbName    = uriParts[2]
```
we have too much URI, we should have mechanic to handle it better

hieunmce commented on backend/auth/auth.go:
```diff
@@ -57,86 +58,96 @@ func (jwt *JWT) VerifierHandler() func(http.Handler) http.Handler {
 }
 
+// parse uri => type, data, ok
+func parseURI(uri string) (int, []string, bool) {
```
This function is not cool (because the return paramter `int` is not clear what it is it will make ppl confuse also we can use multiple return function instead of array

roycephan committed:
user group permission

- [x] add group to config
- [x] get final permission
- [x] auth user's permission & group permission
- [x] group's permission foreach table
- [x] test for group permission
- [x] crud group

hieunmce commented on backend/auth/auth.go:
```diff
@@ -57,86 +58,95 @@ func (jwt *JWT) VerifierHandler() func(http.Handler) http.Handler {
 	return jwtauth.Verifier(jwt.TokenAuth)
 }
 
+const (
```
We should change this

roycephan committed:
use enum
